File: rb/Irrgarten/Monster.rb
#encoding:UTF-8
require_relative'LabyrinthCharacter'

module Irrgarten
    class Monster < LabyrinthCharacter
        @@INITIAL_HEALTH=5.0

        def initialize(name, intelligence, strength)
            super(name,intelligence,strength,@@INITIAL_HEALTH)
        end
        
        def attack
            return Dice.intensity(@strength)
        end

        def defend(received_attack)
            is_dead = dead?
            if (!is_dead)
                defensive_energy = Dice.intensity(@intelligence)
                if (defensive_energy < received_attack)
                    got_wounded
                    is_dead = dead?
                end
            end
            is_dead
        end

        def to_s
            super
        end

        private

        def got_wounded
            @health-=1
        end
    end
end

File: ExamenP5/Irrgarten/src/irrgarten/Monster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author Sergio Salvador Gil
 */
public class Monster extends LabyrinthCharacter{
    
    private static final int INITIAL_HEALTH=5;
    
    // Métodos públicos de la clase Monster
    
    public Monster(String name, float intelligence, float strength){
        super(name,intelligence,strength,INITIAL_HEALTH);
    }
    
    @Override
    public float attack(){
        return Dice.intensity(this.getStrength());
    }
    
    @Override
    public boolean defend(float receivedAttack){
        boolean isDead= this.dead();
        
        if (!isDead){
            
            float defensiveEnergy = Dice.intensity(this.getIntelligence());
            
            if(defensiveEnergy < receivedAttack){
                this.gotWounded();
                isDead=this.dead();
            }
        }
        return isDead;
    }
    
    @Override
    public String toString(){
        return super.toString();
    }
}
